src/librustc_typeck/check/_match.rs

```diff
@@ -607,7 +607,12 @@ https://doc.rust-lang.org/reference/types.html#trait-objects");
 
         // If there are no arms, that is a diverging match; a special case.
         if arms.is_empty() {
-            self.diverges.set(self.diverges.get() | Diverges::Always);
+            let diverges = match self.diverges.get() {
```

Comment:
Why does this convert `UnwarnedAlways` to `Always`? This looks like something that needs to be clarified.

Reply:
At the moment, this is special casing to keep the warning behaviour the same as before. I would prefer not to special-case, but I wanted to know what the feelings were about changing the unreachable code warnings.

Reply:
is this comment out of date now?

Reply:
Yes: I replaced `UnwarnedAlways` with the feature gating.